src/main/kotlin/org/jitsi/jicofo/jibri/JibriSipGateway.kt

```diff
@@ -0,0 +1,162 @@
+import org.jitsi.utils.logging2.Logger
+import org.jitsi.xmpp.extensions.jibri.JibriIq
+import org.jitsi.xmpp.extensions.jibri.SipCallState
+import org.jivesoftware.smack.packet.IQ
+import org.jivesoftware.smack.packet.XMPPError
+import kotlin.collections.HashMap
+import org.jitsi.jicofo.util.ErrorResponse.create as error
+
+/**
+ * A Jibri SIP gateway that manages SIP calls for single conference. Relies on
+ * the information provided by [JibriDetector] to tell whether any Jibris
+ * are currently available and to select one for new [JibriSession]s
+ * (JibriSession does the actual selection).
+ *
+ * @author Pawel Domas
+ */
+class JibriSipGateway(
+    conference: JitsiMeetConferenceImpl,
+    xmppProvider: XmppProvider,
+    jibriDetector: JibriDetector,
+    parentLogger: Logger
+) : BaseJibri(
+    conference,
+    xmppProvider,
+    parentLogger,
+    jibriDetector
+) {
+    /**
+     * Th SIP [JibriSession]s mapped by the SIP address.
+     */
+    private val sipSessions: MutableMap<String, JibriSession> = HashMap()
+
+    /**
+     * Accepts only [JibriIq] with a SIP address (packets without one are handled by JibriRecorder).
+     */
+    override fun acceptType(packet: JibriIq): Boolean = StringUtils.isNotBlank(packet.sipAddress)
+
+    /**
+     * {@inheritDoc}
+     */
+    override fun dispose() {
+        try {
+            sipSessions.values.forEach { it.stop(null) }
+        } finally {
+            sipSessions.clear()
+        }
+        super.dispose()
+    }
+
+    override fun getJibriSessionForMeetIq(iq: JibriIq): JibriSession? = sipSessions[iq.sipAddress]
+    override fun getJibriSessions(): List<JibriSession> = ArrayList(sipSessions.values)
+
+    override fun handleStartRequest(iq: JibriIq): IQ =
+        if (StringUtils.isNotBlank(iq.sipAddress)) {
+            val sessionId = generateSessionId()
+            val jibriSession = JibriSession(
+                this,
+                conference.roomName,
+                iq.from,
+                config.pendingTimeout.seconds,
+                config.numRetries,
+                connection,
+                jibriDetector,
+                false,
+                iq.sipAddress,
+                iq.displayName, null, null, sessionId, null,
+                logger
+            )
+            sipSessions[iq.sipAddress] = jibriSession
+            try {
+                jibriSession.start()
+                logger.info("Started Jibri session")
+                JibriIq.createResult(iq, sessionId)
+            } catch (exc: StartException) {
+                val reason = exc.message
+                logger.warn("Failed to start a Jibri session: $reason", exc)
+                sipSessions.remove(iq.sipAddress)
+                when (exc) {
+                    is AllBusy -> error(iq, XMPPError.Condition.resource_constraint, "all Jibris are busy")
+                    is NotAvailable -> error(iq, XMPPError.Condition.service_unavailable, "no Jibri instances available")
```

Comment:
Elsewhere you changed this message to "no Jibris available" I think?

Reply:
I think that changed in a later commit. I tried to make them all consistent now.